### PR TITLE
PMM-9068: Test SMTP config

### PR DESCRIPTION
**Pull request by oter**
[PMM-9068](https://jira.percona.com/browse/PMM-9068)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/2158

- [ ] https://github.com/percona/pmm/pull/798
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/server/server.go [hunk 1]
```diff
@@ -737,6 +741,44 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
...
+	var errInvalidArgument *models.ErrInvalidArgument
+	err := s.emailer.Send(ctx, settings, req.EmailTo)
+	switch {
+	case err == nil:
+	case errors.As(err, &errInvalidArgument):
+		return nil, status.Error(codes.InvalidArgument,
+			fmt.Sprintf("Cannot send email: %s.", errInvalidArgument.Details))
+	default:
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Cannot send email: %s.", err.Error()))
+	}
```
Comment (artemgavrilov) [inline services/server/server.go:777]: IMHO that looks much more simple:
```suggestion
	if err != nil {
		if errors.As(err, &errInvalidArgument) {
			return nil, status.Errorf(codes.InvalidArgument, "Cannot send email: %s.", errInvalidArgument.Details)
		}

		return nil, status.Errorf(codes.Internal, "Cannot send email: %s.", err.Error())
	}
```

**Review by artemgavrilov — CHANGES_REQUESTED** [review]

File: services/server/server.go [hunk 2]
```diff
@@ -737,6 +741,44 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
...
+		Hello:      req.Settings.Hello,
+		Username:   req.Settings.Username,
+		Password:   req.Settings.Password,
+		Identity:   req.Settings.Identity,
+		Secret:     req.Settings.Secret,
+		RequireTLS: req.Settings.RequireTls,
+	}
+
+	if err := settings.Validate(); err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Invalid argument: %s.", err.Error()))
```
[the remark this answers is not in the document]
Comment (oter) [reply services/server/server.go:761]: done

File: services/server/server.go (hunk 1 above)
Comment (oter) [reply services/server/server.go:777]: done

File: services/alertmanager/emailer.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	tmpl, err := template.FromGlobs()
+	if err != nil {
+		return err
+	}
+	tmpl.ExternalURL, err = url.Parse("https://example.com")
+	if err != nil {
+		return err
+	}
+
+	alertmanagerEmail := email.New(emailConfig, tmpl, kitlog.NewNopLogger())
```
[the remark this answers is not in the document]
Comment (oter) [reply services/alertmanager/emailer.go]: done

**Review by artemgavrilov — APPROVED** [review]

**Review by yurkovychv — APPROVED** [review]

**Pull request merged**